### PR TITLE
[CS-4015]: Lock app keeping navigation state

### PR DESCRIPTION
**Pull request by danibonilha**
### Description

This PR fixes the missing navigation state issue while locking the app, it also updates the react-native types which were mismatched, and in other to set the proper listeners needed to be updated, bc the older implementation is being deprecated. 

~~UPDATE: it's a draft, because I identify that the initial flows are not navigating automatically.  🥲~~

UPDATE: it's ready to review, the initial flows are still funky, but it will be handled on a diff PR

### Checklist

- [x] Tested on iOS
- [x] Tested on Android

### Screenshots

Simulator Screen Recording - iPhone 11 - 2022-06-15 at 18 20 09

### CODE REVIEW

**Review by danibonilha — COMMENTED** [review]
.

File: cardstack/src/navigation/tabBarNavigator.tsx [hunk 1]
```diff
@@ -180,8 +179,10 @@ const useNavigationAuth = () => {
   }, []);
 
   useEffect(() => {
-    loadAuthInfo();
-  }, [loadAuthInfo, isAuthorized]);
+    if (!hasPin || isFirstMount.current) {
+      loadAuthInfo();
```
Comment (oidoug) [inline cardstack/src/navigation/tabBarNavigator.tsx:183]: I wonder if `loadAuthInfo` will not be called twice, once `isFirstMount` is true and then if `hasPin` is false.

**Review by oidoug — APPROVED** [review]

**Review by paulinhapenedo — APPROVED** [review]
It's always nice to see unused files being removed and lodash references removed 👏

File: cardstack/src/navigation/tabBarNavigator.tsx (hunk 1 above)
Comment (danibonilha) [reply cardstack/src/navigation/tabBarNavigator.tsx:183]: There's still improvements to be done, the right way is to call just once on init, and use something else to trigger the reload once we have a pin

**Pull request merged**